test/e2e/tests/dapp-interactions/permissions.spec.js

```diff
@@ -28,19 +28,15 @@ describe('Permissions', function () {
           tag: 'button',
         });
 
-        await driver.waitUntilXWindowHandles(3);
-        const windowHandles = await driver.getAllWindowHandles();
-        const extension = windowHandles[0];
-        await driver.switchToWindowWithTitle(
-          WINDOW_TITLES.Dialog,
-          windowHandles,
```

Comment:
all this extra logic is not needed. switchToWindowWithTitle already waits for the dialog and switches when its ready